rskj-core/src/main/java/co/rsk/peg/bitcoin/BitcoinUtils.java

```diff
@@ -90,9 +90,23 @@ public static Script createBaseP2SHInputScriptThatSpendsFromRedeemScript(Script
         return outputScript.createEmptyInputScript(null, redeemScript);
     }
 
+    public static Script getRedeemScriptFromP2SHInputScript(Script inputScript) {
```

Comment:
tests?

Reply:
updated

Reply:
Looking a bit more into this, we already have a method here `extractRedeemScriptFromInput`. Why not use that one passing the input directly? Instead of extracting the input script and then calling this

Reply:
could be, i find it weird for this method to return an optional. I think it should return the script or an exception, wdyt?

Reply:
I think the Optional gives a bit more flexibility, it get the redeem script if there is one. Then you can do all kinds of functional stuff with the optional return.

Reply:
updated